### ISSUE TITLE
Ensure that language tags are available in PDF export (digital)

### ISSUE THREAD

**Issue by MichSpieler** [pressbooks/pressbooks#4263]
A [forum user has reported](https://pressbooks.community/t/language-tags-missing-from-pdf-exports/4456/2) that language tags don't appear consistently in PDF exports.

> We are using the Prince pdf export routine, and, while producing a series of books with a mix of languages, our digital publishing specialist discovered that language tags from the web book do not appear in the pdf exports. These are a required accessibility feature, and because there are thousands of them in these texts, this is not a remediation that we can manage by hand.

> I notice that, oddly, footnotes with `<span lang=”fr”>` are added in the pdf correctly (e.g., the span shows up as a tag and that tag has the language as a property) Pressbooks uses the XHTML for the source document for Prince. I notice that in the XHTML document, footnotes have `<span lang=”fr”>` while other places in the document have `<span xml:lang=”fr”>` or `<p xml:lang=”fr”>` and, for some reason, Prince is not recognizing the xml:lang property but is recognizing the lang property.
>
>After consulting the Prince documentation, I get a few leads. To cover my bases, I add @namespace xml “``http://www.w3.org/XML/1998/namespace”``; to the pdf css, with no effect, and then tell prince that the source of the digital pdf is xml by adding $prince->setInputType(‘xml’); to the Prince wrapper for digital pdfs. This does cause the PDF to generate with the appropriate language property in all of the places I had `<span lang=”fr”>` as well as other places like `<p xml:lang=”fr”>` that were rendering as xml:lang in the xhtml.

When reproducing this issue I found the following in the PDF preview ([PDF preview of chapter on staging](https://staging.pressbooks.network/bugreprobook/format/xhtml?debug=prince#chapter-language-tags-in-pdf-en-francais))
- Language tags in chapter titles are present in the format `<span lang="&quot;fr&quot;" data-ref="8cb37424-3b45-4775-a7eb-f76b904d276f">en français</span>` (there's an additional set of "")
- Language tags in regular text are present in the format `<span xml:lang="fr" data-ref="99ea880a-d58e-4896-9ffb-8f688cfb1992">je change pour le français</span>`
- Language tags in footnotes are present in the format `<span lang="fr" data-ref="0936044d-ca1e-4a67-8bd4-ed044d07eebb">je change pour le français</span>`
- Language tags in image captions are present in the format `<span xml:lang="fr" data-ref="7ed322bd-dcb7-4298-b194-2841c76c98da">et ça c’est en français</span>`

**Comment by alexdryden** [comment]
I'm writing here with some additional context on finding the language tags in Adobe, because it is not obvious: 

Make sure that the Accessibility Tags are in the side panel: View->Show/Hide->Side panels->Accessibility tags
<img width="1165" height="300" alt="Image" src="https://github.com/user-attachments/assets/a9ef3a59-7291-4334-9861-f5cf8cafaa5a" />
Open the Accessibility Tags panel and go to a tag that should have/contain a language tag. Click to open the nested tree until you arrive at the text that should be tagged:

<img width="450" height="488" alt="Image" src="https://github.com/user-attachments/assets/56874e8f-c2c4-4b44-aaa5-41d6f39e3084" />

Right click and select properties:


<img width="450" height="555" alt="Image" src="https://github.com/user-attachments/assets/ab8f925b-8d38-41a7-b5e2-7d4f3c5cfcc1" />

You should see the tagged language

<img width="500" height="297" alt="Image" src="https://github.com/user-attachments/assets/a76a0ae4-9e0d-4faf-adf3-42d2f02a419e" />

For any of the ones where you find `xml:lang='fr'` in the xhtml, I do not find a tag in the pdf. When I tell Prince to that the document is an xml, these are parsed correctly and I do see the tag in the pdf.

**Comment by alexdryden** [comment]
Adding PR to help highlight the change I made and to facilitate further debugging. If this is a solution you like, I'd be happy to add test coverage.

**Comment by arzola** [comment]
Ok this was tricky to solve, a combination of Outdated prince + htmlawed escaping

This would require at least `prince 15.4.1` or `DocRaptor > 10.1` pipeline

https://github.com/pressbooks/pressbooks/pull/4414

Definitely it was not a 2, I spent several hours.

**Comment by arzola** [comment]
We need Docraptor to be bumped to support accessibility tags

I updated Docraptor pipeline to use Prince 16 and it works, this could also be highly beneficial to have better support for Mathml.

**Comment by arzola** [comment]
In order to have this feature available for our clients we need to bump our pipeline (10 - 11) to use Prince 16, we need to double check what that means for our exports, but I also consider this could be highly beneficial to explore it considering it has JavaScript ES6 support and tons of accessibility fixes that maybe could help us to improve also this https://github.com/pressbooks/pb-mathjax/issues/128#issuecomment-4005301770

**Comment by arzola** [comment]
Blocked until we complete #4418

**Comment by tw77** [comment]
Working as expected on dev networks now with a digital PDF export. Language tags appear in Adobe as described above, including in regular text, an image caption, and a footnote. Ready for release to production.

**Comment by arzola** [comment]
This was deployed to production networks
